purlcli.py

```diff
@@ -0,0 +1,59 @@
+import json
+
+import click
+import requests
+
+
+@click.group()
+def purlcli():
+    """
+    Return information from a PURL.
+    """
+
+
+@purlcli.command(name="validate")
+@click.option(
+    "--purl",
+    "purls",
+    multiple=True,
+    required=False,
+    help="PackageURL or PURL.",
+)
+@click.option(
+    "--output",
+    type=click.File(mode="w", encoding="utf-8"),
+    required=True,
+    default="-",
+    help="Write validation output as JSON to FILE.",
+)
+@click.option(
+    "--file",
+    type=click.File(mode="r", encoding="utf-8"),
+    required=False,
+    help="Read a list of PURLs from a FILE, one per line.",
+)
+def validate(purls, output, file):
+    """
+    Check the syntax of one or more PURLs.
+    """
+    if (purls and file) or not (purls or file):
+        raise click.UsageError("Use either purls or file but not both.")
+    api_query = "https://public.purldb.io/api/validate/"
+    validated_purls = []
+    if file:
+        purls = file.read().splitlines(False)
+
+    for purl in purls:
```

Comment:
IMHO you could extract the whole code block from line 46 down to line 53 in a new function called `def validate_purls(purls)` that will accept a list of purls and returns a list of `validated_purls`
This fill have no dependency on the CLI, files, and the output and will be therefore easy to test. And then write a few tests on this function,